neonvm-runner/cmd/main.go

```diff
@@ -669,6 +673,91 @@ func main() {
 	}
 }
 
+type NetworkMonitoringMetrics struct {
+	IngressBytes, EgressBytes       prometheus.Counter
+	Errors                          *prometheus.CounterVec
+	IngressBytesRaw, EgressBytesRaw uint64 // Absolute values to calc increments for Counters
+}
+
+func NewMonitoringMetrics(reg *prometheus.Registry) *NetworkMonitoringMetrics {
+	m := &NetworkMonitoringMetrics{
+		IngressBytes: util.RegisterMetric(reg, prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Name: "runner_vm_ingress_bytes",
+				Help: "Number of bytes received by the VM from the open internet",
+			},
+		)),
+		EgressBytes: util.RegisterMetric(reg, prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Name: "runner_vm_egress_bytes",
+				Help: "Number of bytes sent by the VM to the open internet",
+			},
+		)),
+		IngressBytesRaw: 0,
+		EgressBytesRaw:  0,
+		Errors: util.RegisterMetric(reg, prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "runner_vm_network_fetch_errors_total",
+				Help: "Number of errors while fetching network monitoring data",
+			},
+			[]string{"cause"},
```

Comment:
I'm worried about label explosion here, especially because this metric will already be collected per-pod. Let's just make this a regular `Counter`, not `CounterVec` ?